<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>3. Preparation of the reference sequence &mdash; Archaeogenetics in Porto 0.0.1 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="4. Alignment of the reads to the reference sequence" href="4_ReadsMapping_v2_2022.html" />
    <link rel="prev" title="2. Quality filtering of reads" href="2_ReadsFiltering_2022.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> Archaeogenetics in Porto
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="1_ListTools.html">1. List of Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_ReadsFiltering_2022.html">2. Quality filtering of reads</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">3. Preparation of the reference sequence</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#index-the-reference-sequence-with-bwa">3.1. Index the reference sequence with BWA</a></li>
<li class="toctree-l2"><a class="reference internal" href="#create-a-reference-dictionary">3.2. Create a reference dictionary</a></li>
<li class="toctree-l2"><a class="reference internal" href="#index-the-reference-sequence-with-samtools">3.3. Index the reference sequence with Samtools</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="4_ReadsMapping_v2_2022.html">4. Alignment of the reads to the reference sequence</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_SummaryReports.html">5. Create Summary Reports</a></li>
<li class="toctree-l1"><a class="reference internal" href="6_DamageAnalysis.html">6. Damage analysis and quality rescaling of the BAM file</a></li>
<li class="toctree-l1"><a class="reference internal" href="7_VariantsCall_2022.html">7. Variant calling and visualization</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Archaeogenetics in Porto</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li><span class="section-number">3. </span>Preparation of the reference sequence</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/3_PreparationReference_2022.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="preparation-of-the-reference-sequence">
<h1><span class="section-number">3. </span>Preparation of the reference sequence<a class="headerlink" href="#preparation-of-the-reference-sequence" title="Permalink to this headline"></a></h1>
<p>After quality-control and pre-processing of the reads (trimming of adapter sequences and merging of paired reads), we will align the merged and trimmed reads to the reference sequence of interest, available in the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq NCBI</a> database. The first thing to do is the prepare the reference sequence for the alignment procedure.</p>
<blockquote>
<div></div></blockquote>
<section id="index-the-reference-sequence-with-bwa">
<h2><span class="section-number">3.1. </span>Index the reference sequence with BWA<a class="headerlink" href="#index-the-reference-sequence-with-bwa" title="Permalink to this headline"></a></h2>
<p>To align the reads to the reference sequence we will use the program <a class="reference external" href="https://github.com/lh3/bwa">BWA</a>, in particular the <code class="docutils literal notranslate"><span class="pre">BWA</span> <span class="pre">aln</span></code> algorithm. BWA first needs to construct the <strong>FM-index</strong> for the reference genome, with the command <code class="docutils literal notranslate"><span class="pre">BWA</span> <span class="pre">index</span></code>. FM-indexing in Burrows-Wheeler transform is used to efficiently find the number of occurrences of a pattern within a compressed text, as well as locate the position of each occurrence. It is an essential step for querying the DNA reads to the reference sequence. This command generates five files with different extensions: <code class="docutils literal notranslate"><span class="pre">amb</span></code>, <code class="docutils literal notranslate"><span class="pre">ann</span></code>, <code class="docutils literal notranslate"><span class="pre">bwt</span></code>, <code class="docutils literal notranslate"><span class="pre">pac</span></code>, <code class="docutils literal notranslate"><span class="pre">sa</span></code>.</p>
<blockquote>
<div></div></blockquote>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bwa</span> <span class="n">index</span> <span class="o">-</span><span class="n">a</span> <span class="ow">is</span> <span class="n">reference</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The option <code class="docutils literal notranslate"><span class="pre">-a</span></code> indicates the algorithm to use for constructing the index. For genomes smaller than &lt; 2 Gb use the <code class="docutils literal notranslate"><span class="pre">is</span></code> algorithm. For larger genomes (&gt;2 Gb), use the <code class="docutils literal notranslate"><span class="pre">bwtsw</span></code> algorithm.</p>
</div>
</section>
<section id="create-a-reference-dictionary">
<h2><span class="section-number">3.2. </span>Create a reference dictionary<a class="headerlink" href="#create-a-reference-dictionary" title="Permalink to this headline"></a></h2>
<p>A dictionary file (<code class="docutils literal notranslate"><span class="pre">dict</span></code>) is necessary to run later in the pipeline <code class="docutils literal notranslate"><span class="pre">GATK</span> <span class="pre">RealignerTargetCreator</span></code>. A sequence dictionary contains the sequence name, sequence length, genome assembly identifier, and other information about the sequences. To create the <code class="docutils literal notranslate"><span class="pre">dict</span></code> file we use <a class="reference external" href="https://broadinstitute.github.io/picard/">Picard</a>.</p>
<blockquote>
<div></div></blockquote>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>picard CreateSequenceDictionary R= referece.fasta O= reference.dict

In some environments we can call ``Picard`` just by typing the program name. In other environments (including this server) you may have to call Picard by providing the full path to the java file (``jar``) of the program. Here, the path is: ``java -jar /home/aurochs/Software/picard/picard.jar``
::

  java -jar /home/aurochs/Software/picard/picard.jar CreateSequenceDictionary R= referece.fasta O= ref.dict
</pre></div>
</div>
</section>
<section id="index-the-reference-sequence-with-samtools">
<h2><span class="section-number">3.3. </span>Index the reference sequence with Samtools<a class="headerlink" href="#index-the-reference-sequence-with-samtools" title="Permalink to this headline"></a></h2>
<p>The reference sequence has to be indexed in order to run later in the pipeline <code class="docutils literal notranslate"><span class="pre">GATK</span> <span class="pre">IndelRealigner</span></code>. To do that, we will use <a class="reference external" href="http://www.htslib.org/">Samtools</a>, in particular the tool <code class="docutils literal notranslate"><span class="pre">samtools</span> <span class="pre">faidx</span></code>, which enables efficient access to arbitrary regions within the reference sequence. The index file typically has the same filename as the corresponding reference sequece, with the extension <code class="docutils literal notranslate"><span class="pre">fai</span></code> appended.</p>
<blockquote>
<div></div></blockquote>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">faidx</span> <span class="n">reference</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="2_ReadsFiltering_2022.html" class="btn btn-neutral float-left" title="2. Quality filtering of reads" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="4_ReadsMapping_v2_2022.html" class="btn btn-neutral float-right" title="4. Alignment of the reads to the reference sequence" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Claudio Ottoni.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>
    <!-- Theme Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXX-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'UA-XXXXXXX-1', {
          'anonymize_ip': false,
      });
    </script> 

</body>
</html>