

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>7. Variant calling and visualization &mdash; Archaeogenetics 0.0.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript">
          var DOCUMENTATION_OPTIONS = {
              URL_ROOT:'./',
              VERSION:'0.0.1',
              LANGUAGE:'None',
              COLLAPSE_INDEX:false,
              FILE_SUFFIX:'.html',
              HAS_SOURCE:  true,
              SOURCELINK_SUFFIX: '.txt'
          };
      </script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="6. Damage analysis and quality rescaling of the BAM file" href="DamageAnalysis.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Archaeogenetics
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="ListTools.html">1. List of Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="ReadsFiltering.html">2. Quality filtering of reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="PreparationReference.html">3. Preparation of the reference sequence</a></li>
<li class="toctree-l1"><a class="reference internal" href="ReadsMapping_v2.html">4. Alignment of the reads to the reference sequence</a></li>
<li class="toctree-l1"><a class="reference internal" href="SummaryReports.html">5. Create Summary Reports</a></li>
<li class="toctree-l1"><a class="reference internal" href="DamageAnalysis.html">6. Damage analysis and quality rescaling of the BAM file</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">7. Variant calling and visualization</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#variants-calling">7.1. Variants calling</a></li>
<li class="toctree-l2"><a class="reference internal" href="#variants-visualization">7.2. Variants visualization</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Archaeogenetics</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>7. Variant calling and visualization</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/VariantsCall.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="variant-calling-and-visualization">
<h1>7. Variant calling and visualization<a class="headerlink" href="#variant-calling-and-visualization" title="Permalink to this headline">¶</a></h1>
<p>Once the reads are aligned and the data authenticated through post-mortem damage analysis, we can analyse the variant positions in the samples against the reference sequence.</p>
<div class="section" id="variants-calling">
<h2>7.1. Variants calling<a class="headerlink" href="#variants-calling" title="Permalink to this headline">¶</a></h2>
<p>We will use two common tools for varaints calling: <em>samtools mpileup</em> in combination with <em>bcftools</em>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">mpileup</span> <span class="o">-</span><span class="n">B</span> <span class="o">-</span><span class="n">ugf</span> <span class="n">reference</span><span class="o">.</span><span class="n">fasta</span> <span class="n">filename</span><span class="o">.</span><span class="n">final</span><span class="o">.</span><span class="n">sort</span><span class="o">.</span><span class="n">rescaled</span><span class="o">.</span><span class="n">bam</span> <span class="o">|</span> <span class="n">bcftools</span> <span class="n">call</span> <span class="o">-</span><span class="n">vmO</span> <span class="n">z</span> <span class="o">-</span> <span class="o">&gt;</span> <span class="n">filename</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>The detected genetic variations will be stored in the generated vcf file <cite>output.vcf.gz</cite>. These genetic variations can be filtered according to some criteria using bcftools:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bcftools</span> <span class="nb">filter</span> <span class="o">-</span><span class="n">O</span> <span class="n">z</span> <span class="o">-</span><span class="n">o</span> <span class="n">filename</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">s</span> <span class="n">LOWQUAL</span> <span class="o">-</span><span class="n">i</span><span class="s1">&#39;%QUAL&gt;19&#39;</span> <span class="n">filename</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>other options can be added when using bcftools filter:</p>
<blockquote class="last">
<div><table border="1" class="docutils">
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Option</th>
<th class="head">Function</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><em>-g, –SnpGap</em> &lt;int&gt;</td>
<td>filter SNPs within &lt;int&gt; base pairs of an indel</td>
</tr>
<tr class="row-odd"><td><em>-G, –IndelGap</em> &lt;int&gt;</td>
<td>filter clusters of indels separated by &lt;int&gt; or fewer base pairs allowing only one to pass</td>
</tr>
</tbody>
</table>
</div></blockquote>
</div>
<p>Instead of samtools mpileup and bcftools (or in addition to) we can use <em>gatk HaplotypeCaller</em>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">java</span> <span class="o">-</span><span class="n">jar</span> <span class="n">GenomeAnalysisTK</span><span class="o">.</span><span class="n">jar</span> <span class="o">-</span><span class="n">T</span> <span class="n">HaplotypeCaller</span> <span class="o">-</span><span class="n">R</span> <span class="n">reference</span><span class="o">.</span><span class="n">fasta</span> <span class="o">-</span><span class="n">I</span> <span class="n">filename</span><span class="o">.</span><span class="n">final</span><span class="o">.</span><span class="n">sort</span><span class="o">.</span><span class="n">rescaled</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">o</span> <span class="n">original</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
<span class="n">java</span> <span class="o">-</span><span class="n">jar</span> <span class="n">GenomeAnalysisTK</span><span class="o">.</span><span class="n">jar</span> <span class="o">-</span><span class="n">T</span> <span class="n">VariantFiltration</span> <span class="o">-</span><span class="n">R</span> <span class="n">reference</span><span class="o">.</span><span class="n">fasta</span> <span class="o">-</span><span class="n">V</span> <span class="n">filename</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">o</span> <span class="n">filename</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">filterName</span> <span class="s1">&#39;Cov3|Qual20&#39;</span> <span class="o">--</span><span class="n">filterExpression</span> <span class="s1">&#39;DP&gt;2&#39;</span><span class="o">--</span><span class="n">filterExpression</span> <span class="s1">&#39;QUAL&gt;19&#39;</span>
</pre></div>
</div>
</div>
<div class="section" id="variants-visualization">
<h2>7.2. Variants visualization<a class="headerlink" href="#variants-visualization" title="Permalink to this headline">¶</a></h2>
<p>To be able to visualize the identified variations in the vcf files in their context, you can use the program IGV that accepts multiple input files formats eg. fasta, bam, vcf and gff. After loading your bam file(s) and the corrsponding vcf file(s), this is what you will see:</p>
<img alt="_images/igv-bam_vcf.png" src="_images/igv-bam_vcf.png" />
<p>As you can see, from the bam file we can see that there is a single nucleotide polymorphism (SNP) that corresponds to a T to C. This SNP is ofcourse represented in the vcf file data session of the igv viewer.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="DamageAnalysis.html" class="btn btn-neutral float-left" title="6. Damage analysis and quality rescaling of the BAM file" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Claudio Ottoni

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
    <!-- Theme Analytics -->
    <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-XXXXXXX-1', 'auto');
    ga('send', 'pageview');
    </script>

    
   

</body>
</html>